### ISSUE TITLE
Marketplace: Show what version of VS code an extension is for in the version history page

### ISSUE THREAD

**Issue by sandy081** [microsoft/vsmarketplace#43]
From @rchiodo

Our extension has this list currently:

![image](https://user-images.githubusercontent.com/19672699/113047104-bd5a6e00-9155-11eb-8ca0-da95e8de6252.png)

Some of these work with insiders and some of these work with VS code. Users frequently try installing the latest version and don't realize why it doesn't work with their version of VS code. 

Some indication of the version a vsix works with would be nice (or some way for us to put it into the version information).

**Comment by RandomFractals** [comment]
is not that information readily available in extension manifest via `engines: {"vscode": "..."}` field, or can be easily extracted from `"devDependencies": { "@types/vscode": "..." ...}` that all extensions have. You just need to use that info and added to version history page.

**Comment by indyspike** [comment]
The "Version History" at least needs the ability to select a code engine version. [Jupyter](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) version history is only listing versions for 1.58+, so trying to download a compatible version for 1.57.x for offline installation is impossible (linked to microsoft/vscode-jupyter#5582).

**Comment by RandomFractals** [comment]
yeah, I agree that information is needed, especially for some of the "Preview" extensions that are published and work only with vscode insiders versions.

**Comment by amtadev** [comment]
Also, the download API should be updated to accept code engine version. Is there an API which returns all the versions of a particular extension which support a particular vs code engine version?

**Comment by dennisl68-castra** [comment]
This is especially important if for some reason I need to use an older version of vscode-jupyter. 
I don't have the faintest about what version of VS Code older stable versions of vscode-jupyter is lock-stepped with.
